src/Disks/getOrCreateDiskFromAST.cpp

```diff
@@ -43,6 +51,9 @@ namespace
             return disk;
         });
 
+        if (!result_disk->isCustomDisk())
```

Comment:
TODO: this actually saves only from a user attempting to create a custom disk with disk name already present in storage configuration in config file, but does not save from creating a custom disk with name same as already existing custom disk. Will fix it.

Reply:
After a second thought: we cannot easily protect against attempt to make custom disk which clashes with another custom disk name but has different disk configuration (it can be done only if having a global map of custom disks and their ast hashes or something like this, I do not want this overcomplication). So I'd better live it as is. 
The most important is not to allow making custom disk with name as existing disk from config, which will be ensured already by this check in line 54.